scripts/buildProtocol.ts

```diff
@@ -113,7 +113,7 @@ class DeclarationsWalker {
     } 
 }
 
-function generateProtocolFile(protocolTs: string, typeScriptServicesDts: string): string {
+function generateProtocolFile(outputFile: string, protocolTs: string, typeScriptServicesDts: string) {

```

Comment:
Correct. The previous behavior was that this would *not* write out the file if it contained compile errors, but it's hard to diagnose why the file isn't compiling if you can't see it. So this now means "generate to disk" instead of "generate in memory" and throws the exception after writing out the file so that you can actually see it to figure out what's wrong